basic_autonomy/test/test_waypoint_generation.cpp

```diff
@@ -231,7 +231,7 @@ namespace basic_autonomy
         ASSERT_EQ(3, waypoint_generation::get_nearest_point_index(points, state));
     }
 
-    TEST(BasicAutonomyTest, get_nearest_index_by_downtrack)
+    TEST(BasicAutonomyTest, get_nearest_index_by_downtrack_with_nonempty_points)
```

Comment:
Optional note: Google Test's documentation discourages underscores in test names. I'm not sure if there's a technical justification or if it's a style guide decision. Either way, we should refactor all test names to use `PascalCase`.

Reply:
As we discussed already, this is interesting and I wasn't aware of this before. I will make a GitHub issue to allow us to track this fix across our repo.

Reply:
Created GitHub issue #2332 for this